components/nimbus/src/enrollment.rs

```diff
@@ -1445,6 +1749,231 @@ mod tests {
+            .iter()
+            .filter(|&e| {
+                matches!(
+                    e.status,
+                    EnrollmentStatus::NotEnrolled {
+                        reason: NotEnrolledReason::FeatureConflict
+                    }
+                )
+            })
+            .count();
+        assert_eq!(
+            1, not_enrolleds,
+            "exactly one enrollment should have NotEnrolled status"
+        );
+
+        let enrolleds = enrollments
+            .iter()
+            .filter(|&e| matches!(e.status, EnrollmentStatus::Enrolled { .. }))
+            .count();
+        assert_eq!(
+            2, enrolleds,
+            "exactly two enrollments should have Enrolled status"
+        );
+
+        log::debug!("events: {:?}", events);
+
+        assert_eq!(
+            2,
+            events.iter().count(),
+            "There should be exactly 2 enrollment_change_events"
+        );
+
+        let enrolled_events = events
+            .iter()
+            .filter(|&e| matches!(e.change, EnrollmentChangeEventType::Enrollment))
+            .count();
+        assert_eq!(
+            2, enrolled_events,
+            "exactly two events should have Enrolled event types"
+        );
+
+        Ok(())
+    }
+
+    #[test]
+    fn test_feature_id_reuse() -> Result<()> {
+        let _ = env_logger::try_init();
+
+        let test_experiments = get_test_experiments();
+        let (nimbus_id, app_ctx, aru) = local_ctx();
+        let evolver = EnrollmentsEvolver::new(&nimbus_id, &aru, &app_ctx);
+        let (enrollments, _) = evolver.evolve_enrollments(true, &[], &test_experiments, &[])?;
+
+        let enrolleds = enrollments
+            .iter()
+            .filter(|&e| matches!(e.status, EnrollmentStatus::Enrolled { .. }))
+            .count();
+        assert_eq!(
+            2, enrolleds,
+            "exactly two enrollments should have Enrolled status"
+        );
+
+        let conflicting_experiment = get_conflicting_experiment();
+        let (enrollments, events) = evolver.evolve_enrollments(
+            true,
+            &test_experiments,
+            &[test_experiments[0].clone(), conflicting_experiment.clone()],
+            &enrollments,
+        )?;
+
+        log::debug!("events = {:?}", events);
+
+        assert_eq!(events.len(), 2);
+
+        // we didn't include test_experiments[1] in next_experiments above,
+        // so it should have been unenrolled...
+        assert_eq!(events[0].experiment_slug, test_experiments[1].slug);
+        assert_eq!(events[0].change, EnrollmentChangeEventType::Unenrollment);
+
+        // ...which will have gotten rid of the thing that otherwise would have
+        // conflicted with conflicting_experiment, allowing it to have now
+        // been enrolled.
+        assert_eq!(events[1].experiment_slug, conflicting_experiment.slug);
+        assert_eq!(events[1].change, EnrollmentChangeEventType::Enrollment);
+
+        let enrolleds = enrollments
```

Comment:
super nit, but `enrolleds` is not the most descriptive, suggest `enrolled_count`.